instrumentation/lettuce/lettuce-4.0/src/main/java/io/opentelemetry/auto/instrumentation/lettuce/v4_0/InstrumentationPoints.java

```diff
@@ -52,45 +47,39 @@ public static void afterCommand(
       final AsyncCommand<?, ?, ?> asyncCommand) {
     final Span span = spanWithScope.getSpan();
     if (throwable != null) {
-      LettuceClientDecorator.DECORATE.onError(span, throwable);
-      LettuceClientDecorator.DECORATE.beforeFinish(span);
-      span.end();
+      LettuceDatabaseClientTracer.TRACER.endExceptionally(span, throwable);
     } else if (expectsResponse(command)) {
       asyncCommand.handleAsync(
           (value, ex) -> {
-            if (ex instanceof CancellationException) {
+            if (ex == null) {
+              LettuceDatabaseClientTracer.TRACER.end(span);
+            } else if (ex instanceof CancellationException) {
               span.setAttribute("db.command.cancelled", true);
+              LettuceDatabaseClientTracer.TRACER.end(span);
             } else {
-              LettuceClientDecorator.DECORATE.onError(span, ex);
+              LettuceDatabaseClientTracer.TRACER.endExceptionally(span, ex);
             }
-            LettuceClientDecorator.DECORATE.beforeFinish(span);
-            span.end();
             return null;
           });
     } else {
       // No response is expected, so we must finish the span now.
-      LettuceClientDecorator.DECORATE.beforeFinish(span);
-      span.end();
+      LettuceDatabaseClientTracer.TRACER.end(span);
     }
     spanWithScope.closeScope();
-    // span may be finished by handleAsync call above.
   }
 
   public static SpanWithScope beforeConnect(final RedisURI redisURI) {
-    final Span span =
-        LettuceClientDecorator.TRACER.spanBuilder("CONNECT").setSpanKind(CLIENT).startSpan();
-    LettuceClientDecorator.DECORATE.afterStart(span);
-    LettuceClientDecorator.DECORATE.onConnection(span, redisURI);
-    return new SpanWithScope(span, currentContextWith(span));
+    final Span span = LettuceDatabaseClientTracer.TRACER.startSpan("CONNECT", redisURI, null);
```

Comment:
I think we do need to solve @trask's suggestion, probably by having methods with different names. It's super-confusing to see methods used in the same file with different ordering, looked like a bug at first.

Reply:
Agreed this is bad. But this IS `startSpan` method... I will take another look if I can improve it somehow...

Reply:
This is Java, so I think it's pretty idiomatic when running into issues with overloads, instead of reordering arguments, to use `with`, e.g., `startSpanWithQuery` / `startSpanWithName`. If we expect one of these to be the vast majority, than that one can be shortened to `startSpan`. What do you think?

Reply:
I have reworked this a little, PTAL.